src/core/class_db.cpp

```diff
@@ -341,13 +342,16 @@ void ClassDB::initialize(GDExtensionInitializationLevel p_level) {
 }
 
 void ClassDB::deinitialize(GDExtensionInitializationLevel p_level) {
-	for (const std::pair<StringName, ClassInfo> pair : classes) {
-		const ClassInfo &cl = pair.second;
+	for (size_t i = class_register_order.size() - 1; i >= 0; i--)
```

Comment:
I get a segfault when testing with this PR - the cause is the for loop here.

On my system at least, `size_t` is unsigned. This loop will run so long as `i >= 0`, which means that after i is `0` and it decrements again, it'll wrap around to the max value (rather than stopping the loop), and try to access beyond the length of the vector. And if the segfault didn't kill it, it would loop forever. :-)

If I change the loop to look like this:

```
for (std::vector<StringName>::reverse_iterator i = class_register_order.rbegin(); i != class_register_order.rend(); ++i) {
		const StringName &name = *i;
```

... then there is no segfault, and the bug this is attempting to fix is fixed!

Reply:
Added that change. Forgot to consider the case of no registered custom classes!